### PR TITLE
mgr/dashboard: fix ceph-users api doc

### PR DESCRIPTION
**Pull request by nizamial09**
Fixes: https://tracker.ceph.com/issues/67224

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
@zdover23  The docs are updated and the examples are properly reflected to show how the request body should look like:https://ceph--58951.org.readthedocs.build/en/58951/mgr/ceph_api/#post--api-cluster-user

**Review by zdover23 — APPROVED** [review]

**Review by Pegonzal — COMMENTED** [review]
Thanks @nizamial09 for taking care of this, just one thing from my side

File: src/pybind/mgr/dashboard/controllers/ceph_users.py [hunk 1]
```diff
@@ -39,19 +39,11 @@ def _run_auth_command(command: str, *args, **kwargs):
 
     @staticmethod
     def user_list(_):
-        """
-        Get list of ceph users and its respective data
-        """
```
[the remark this answers is not in the document]
Comment (zdover23) [reply src/pybind/mgr/dashboard/controllers/ceph_users.py:44]: I will rewrite any strings that you guys decide need to be rewritten. Just indicate them. (I won't touch the yaml directly. I'll just comment here.)

File: src/pybind/mgr/dashboard/openapi.yaml [hunk 2]
```diff
@@ -3711,26 +3709,37 @@ paths:
...
       security:
       - jwt: []
-      summary: Get Ceph Users
+      summary: Get list of ceph users and its respective data
       tags:
       - Cluster
     post:
-      description: "\n        Add a ceph user with its defined capabilities.\n   \
-        \     :param user_entity: Entity to change\n        :param capabilities: List\
-        \ of capabilities to add to user_entity\n        "
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/openapi.yaml:3720]: I still think that we should keep the method description (not the param).

Comment (zdover23) [reply src/pybind/mgr/dashboard/openapi.yaml:3720]: Re line 3712: s/respective/associated/

**Comment by zdover23** [comment]
jenkins retest api

File: src/pybind/mgr/dashboard/controllers/ceph_users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/pybind/mgr/dashboard/controllers/ceph_users.py]: makes sense, i addressed the comments.

File: src/pybind/mgr/dashboard/openapi.yaml (hunk 2 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/openapi.yaml:3720]: done!

**Comment by nizamial09** [comment]
sorry for missing this out for a while, but this should be now up-to-date. I've kept the description (minus the params) and fixed the other area.

**Comment by zdover23** [comment]
jenkins test api

**Comment by zdover23** [comment]
jenkins test make check

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/59140 - Squid backport
https://github.com/ceph/ceph/pull/59141 - Reef backport
https://github.com/ceph/ceph/pull/59142 - Quincy backport

**Pull request merged**